crates/project/src/terminals.rs

```diff
@@ -379,17 +369,60 @@ fn prepare_ssh_shell(
     // todo(windows)
     #[cfg(not(target_os = "windows"))]
     std::fs::set_permissions(ssh_path, smol::fs::unix::PermissionsExt::from_mode(0o755))?;
-    let path = format!(
-        "{}:{}",
-        tmp_dir.to_string_lossy(),
-        env.get("PATH")
-            .cloned()
-            .or(env::var("PATH").ok())
-            .unwrap_or_default()
-    );
-    env.insert("PATH".to_string(), path);
+
+    prepare_env_paths(env, tmp_dir)?;
 
     let mut args = shlex::split(&ssh_command).unwrap_or_default();
     let program = args.drain(0..1).next().unwrap_or("ssh".to_string());
     Ok(Shell::WithArguments { program, args })
 }
+
+fn prepare_env_paths(env: &mut HashMap<String, String>, add_path: &Path) -> anyhow::Result<()> {
```

Comment:
I'm not sure what the best name for this function is (since it does quite a lot), but I think this is better:

```suggestion
fn add_environment_path(env: &mut HashMap<String, String>, new_path: &Path) -> anyhow::Result<()> {
```